Makefile

```diff
@@ -0,0 +1,116 @@
+#! /usr/bin/env make -f
+
+# This variable contains the first goal that matches any of the listed goals
+# here, else it contains an empty string. The net effect is to filter out
+# whether this current run of `make` requires a Python virtual environment.
+NEED_VENV := $(or \
+  $(findstring upgrade,$(MAKECMDGOALS)), \
+  $(findstring requirements,$(MAKECMDGOALS)), \
+  $(findstring all,$(MAKECMDGOALS)), \
+  $(findstring quick-check,$(MAKECMDGOALS)), \
+  $(findstring check,$(MAKECMDGOALS)), \
+  $(findstring test,$(MAKECMDGOALS)), \
+  $(findstring dist,$(MAKECMDGOALS)), \
+  $(findstring bdist-wheel,$(MAKECMDGOALS)), \
+  $(findstring sdist,$(MAKECMDGOALS)), \
+  $(findstring docs,$(MAKECMDGOALS)) \
+)
+ifeq ($(NEED_VENV),)
+  # None of the current goals requires a virtual environment.
+else
+  ifeq ($(origin VIRTUAL_ENV),undefined)
+    $(warning No Python virtual environment found, proceeding anyway)
+  else
+    PACKAGE_VERSION=$(shell python -c 'import package; print(package.__version__)')
+  endif
+endif
+
+# Create a virtual environment, either for Python3.10 (default) or using
+# the Python interpreter specified in the PYTHON environment variable.
+.PHONY: venv
+venv:
+	if [ ! -z "${VIRTUAL_ENV}" ]; then \
+	  echo "Found an activated Python virtual environment, exiting" && exit 1; \
+	fi
+	if [ -z "${PYTHON}" ]; then \
+	  echo "Creating virtual envirnoment in .venv/ for python3.10" \
+	  python3.10 -m venv --upgrade-deps .venv; \
+	else \
+	  echo "Creating virtual envirnoment in .venv/ for ${PYTHON}" \
+	  ${PYTHON} -m venv --upgrade-deps .venv; \
+	fi
```

Comment:
You need to add  `&&` to run the commands proceeding `echo`. At the moment `make venv` just echoes strings and does nothing:

```
	if [ -z "${PYTHON}" ]; then \
	  echo "Creating virtual envirnoment in .venv/ for python3.10" \
	  && python3.10 -m venv --upgrade-deps .venv; \
	else \
	  echo "Creating virtual envirnoment in .venv/ for ${PYTHON}" \
	  && ${PYTHON} -m venv --upgrade-deps .venv; \
	fi
```

Reply:
Instead of `&&` I separated the two commands with a `;` because the echo can’t (shouldn’t ever) fail so there’s no need to make the second command conditional on the first one ([docs](https://www.gnu.org/software/bash/manual/bash.html#Lists-of-Commands)).

Commit https://github.com/jenstroeger/python-package-template/pull/74/commits/0ae031343262a69a0d0b65be7b5d8c44d9ced26d.